spec/lib/common/client/middleware/response/response_middleware_spec.rb

```diff
@@ -29,73 +29,185 @@
-        stub.get('mhv-generic-html') { [400, { 'Content-Type' => 'application/html' }, mhv_generic_html] }
-        stub.get('mhv-generic-xml') { [400, { 'Content-Type' => 'application/xml' }, mhv_generic_html] }
+        stub.get("#{fake_host}/ok") { [200, { 'Content-Type' => 'application/json' }, message_json] }
+        stub.get("#{fake_host}/not-found") { [404, { 'Content-Type' => 'application/json' }, four_o_four] }
+        stub.get("#{fake_host}/refill-fail") { [400, { 'Content-Type' => 'application/json' }, i18n_type_error] }
+        stub.get("#{fake_host}/mhv-generic-html") { [400, { 'Content-Type' => 'application/html' }, mhv_generic_html] }
+        stub.get("#{fake_host}/mhv-generic-xml") { [400, { 'Content-Type' => 'application/xml' }, mhv_generic_html] }
+        stub.get("#{fake_host}/mhv-html-503") { [503, { 'Content-Type' => 'application/html' }, mhv_generic_html] }
+        stub.get("#{fake_host}/mhv-xml-503") { [503, { 'Content-Type' => 'application/xml' }, mhv_generic_html] }
       end
     end
   end
 
+  let(:rx_breakers_service) do
+    path = URI.parse('http://host.com').path
+    host = URI.parse('http://host.com').host
+    matcher = proc do |request_env|
+      request_env.url.host == host && request_env.url.path =~ /^#{path}/
+    end
+
+    exception_handler = proc do |exception|
+      if exception.is_a?(Common::Exceptions::BackendServiceException)
+        (500..599).cover?(exception.response_values[:status])
+      else
+        false
+      end
+    end
+
+    Breakers::Service.new(
+      name: 'RX',
+      request_matcher: matcher,
+      exception_handler: exception_handler
+    )
+  end
+
   it 'parses json successfully' do
-    client_response = faraday_client.get('ok')
+    client_response = faraday_client.get("#{fake_host}/ok")
     expect(client_response.body).to be_a(Hash)
     expect(client_response.body.keys).to include(:id, :subject, :category, :body)
     expect(client_response.status).to eq(200)
   end
 
   it 'raises client response error' do
-    message = 'BackendServiceException: {:status=>404, :detail=>"Record Not Found", :code=>"VA900", :source=>"blah"}'
-    expect { faraday_client.get('not-found') }
+    message = 'BackendServiceException: VA900 - Record Not Found'
+    expect { faraday_client.get("#{fake_host}/not-found") }
       .to raise_error do |error|
         expect(error).to be_a(Common::Exceptions::BackendServiceException)
-        expect(error.message)
-          .to eq(message)
-        expect(error.errors.first[:detail])
-          .to eq('Record Not Found')
+        expect(error.message).to eq(message)
+        expect(error.errors.first).to have_attributes(
+          title: 'Operation failed',
+          detail: 'Record Not Found',
+          code: 'VA900',
+          href: nil,
+          id: nil,
+          links: [],
+          meta: nil,
+          source: 'blah',
+          status: '400'
```

Comment:
I don't think it should be a 404 for Record not Found for "Prescription not refillable." But perhaps 422 "Unprocessable Entity" would be better?